packages/api/src/external/hie/set-doc-query-progress.ts

```diff
@@ -193,7 +204,29 @@ export function aggregateDocProgress(hieDocProgresses: DocumentQueryProgress[]):
     {}
   );
 
-  return tallyResults;
+  const defaultCompleteProgress: RequiredProgress = {
+    status: "completed",
+    total: 0,
+    errors: 0,
+    successful: 0,
+  };
+
+  const download: RequiredProgress | undefined = tallyResults.download
+    ? tallyResults.download
+    : existingPatientDocProgress.download
+    ? defaultCompleteProgress
+    : undefined;
+
+  const convert: RequiredProgress | undefined = tallyResults.convert
+    ? tallyResults.convert
+    : existingPatientDocProgress.convert
+    ? defaultCompleteProgress
+    : undefined;
```

Comment:
The reason for the above is in the case where if no type is present then we want to check if the overall progress even had the type if it does then mark it as complete if not just leave it undefined

Reply:
Isn't this a workaround, and shouldn't the tally logic always result in some value for each progress type?

Reply:
Yea your right i added the same logic to the `tallyResults`